### ISSUE TITLE
Improve e2e compatibility test declaration

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#3272]
## Summary

Make it easier to ensure possible test are configured/run for e2e compatibilities. 

## Problem Definition

When reviewing https://github.com/cosmos/ibc-go/pull/3271 I find it somewhat difficult to remember what tests should be run against what versions. It seems very likely we could forget to add a test or version and it would go undetected.

## Proposal

By default, I would want all maintained versions to run against all tests. Any test written should be run by default and excluded if necessary. I believe it should always be the case that a test can be run against version X and every version released after version X. What I mean by this, if a test is introduced in v4.1, I'd prefer to declare what version it was released in and then let our scripts/workflows determine which supported versions it should be run against. 

I think ideally the test itself would declare what version it could be run against, I hesitate to keep a secondary list of tests (unless we ensure the secondary list + exclusions = existing tests)

I understand there's a lot of splitting of testing due to gh actions limit on test runs, but I'd be interested in continuing to explore alternative setups as I find it crucial to ensure every possible test is run. 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by chatton** [comment]
I agree that this can be improved, it can be error prone and easy to forget about a test.

in order to implement this, it is likely that we would need to dynamically generate the json files that are currently manually maintained, which is completely possible (we do this already in order to pick up all tests in the e2e directory for PRs).

The part I am unsure of is how to add metadata about which tests should be running against with other versions. It's definitely possible though as all we need is a binary that outputs json for the desired tests to run.

**Comment by chatton** [comment]
related to #3377
